enhancements/installer/ibm-cloud-for-power-vs-ipi.md

```diff
@@ -0,0 +1,305 @@
+of entry for power optimized workloads.
+
+### Goals
+
+- Provide a way to install OpenShift on Power VS infrastructure using the OpenShift installer with IPI provisioned
+against IBM Cloud APIs.
+- Implement the machine-api controllers to provide scaling and managing the cluster nodes post install.
+
+### Non-Goals
+
+- UPI support. We've already documented how to install on pp64le hardware using UPI.
+
+## Proposal
+
+In order to leverage the Power VS platform, we will make use of the Power VS support in the [IBM Cloud Terraform
+Provider][ibm-cloud-terraform-provider]. This provider will interface with IBM Cloud APIs and ultimately enable the
+OpenShift Installer to provision VM resources using Power VS, that will be used as worker and masters of the cluster.
+Unlike some of the existing providers, this depends on image availability of CoreOS in the Power VS cloud in the OVA
+format. There are currently 2 proposals of how that can be achieved.
+
+The expected workflow would be as follows:
+1. The installer is invoked with credentials for ones IBM Cloud account.
+2. Once invoked, the installer interacts with a terraform provider to create a new Power VS service.
+3. The installer or terraform provider will then import a CoreOS OVA boot image file into the Power VS service from IBM
+Cloud Object Storage, which creates a corresponding boot image.
+4. Once imported the installer can start building the private network for the nodes.
+5. The bootstrap and master nodes will be created on in the private network. These are called Power VS instances, and
+they get created using the CoreOS boot image.
+6. In order to expose this private network and provide load balancing, the IBM Cloud APIs will leverage the IBM Cloud
+[Direct Link][direct-link-website] offering. This will allow for both the ingress load balancer and the cluster nodes to
+talk over the private network on a high-speed link.
+7. Bootstrapping completes as normal.
+8. The workers will be added using the openshift/[cluster-\*machine-\*] controllers and operators based on the openshift
+MAPI model. A Power VS controller needs to be implemented and hooked into the [OpenShift Machine
+Operator][openshift-machine-operator].
+9. The cluster teardown can make use of the existing [go client for Power][power-go-client] via `openshift-install
+destroy cluster`.
+
+### User Stories
+
+- A user of this installer would like to see a cluster running within an hour of kicking off the installer.
+- The expectation is that this is fully automated.
+
+### Implementation Details/Notes/Constraints
+
+#### Existing APIs in the Upstream
+There is a working [implementation of a machine API](https://github.com/openshift-powervs/cluster-api-provider-powervs)
+we've created for testing purposes. Though not ready for upstream, many have been tested and found to be working, such
+as mapi and nodeupdate controller. This was tested with machineset as well, so things look promising.
```

Comment:
Have you tried doing any testing with our E2E suite, might be worth running that against your provider to determine if there are any gaps?

https://github.com/openshift/cluster-api-actuator-pkg

Reply:
sure, WIll do

Reply:
@mkumatag any updates on the actuator e2e results?

Reply:
we have created a jira task for the same - https://issues.redhat.com/browse/MULTIARCH-1345 and blocked by some other infra issues.